rfcs/0036-remove-header-deps-immature-rule/0036-remove-header-deps-immature-rule.md

```diff
@@ -1 +1,32 @@
-In review, see <https://github.com/nervosnetwork/rfcs/pull/240>
+---
+Number: "0036"
+Category: Consensus (Hard Fork)
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-02-07
+---
+
+# Remove Header Deps Immature Rule
+
+## Abstract
+
+This document proposes removing the loading header immature rule.
+
+In the consensus ckb2019, the header dep must reference the block which is 4 epochs ago. After this RFC is activated, the transaction can use any existing blocks in the chain as the header dep.
+
+## Motivation
+
+Header dep is a useful feature for dApps developers because script can use it to read the header of a block in the chain, or verify that an input cell or dep cell is in a specific block in the chain.
+
+The immature rule prevents the usage of header deps in many scenarios because the script must reference the block about 16 hours ago.
+
+The intention of the immature rule is like the cellbase immature rule, a transaction with header deps and all its descendants can be invalidated after a chain reorganization [^1], because the referenced block may be rollbacked.
```

Comment:
I feel that the reading has come to an abrupt end here, only saying that the "Immature Rule" is necessary, but then it does not say why the rule should be abolished. Talking about the problem of chain reorganization, when the rule is abolished, how to solve?  

-- suggestion proposed by @chanhsu001 offline